docs/control-flow_macros.cheatmd

```diff
@@ -0,0 +1,47 @@
+# Control Flow Macros (aka "Blocks": `case`, `if`, `unless`, `cond`, `with`)
+
+Elixir's Kernel documentation refers to these structures as "macros for control-flow".
+We often refer to them as "blocks" in our changelog, which is a much worse name, to be sure.
+
+You're likely here just to see what Styler does, in which case, please [click here to skip](skip) following manifesto on our philosophy regarding the usage of these macros.
+
+## Which Control Flow Macro Should I Use?
+
+The number of "blocks" in Elixir means there are a large number of ways to write semantically equivalent code, often leaving developers [in the dark as to which structure they should use.](https://www.reddit.com/r/elixir/comments/1ctbtcl/i_am_completely_lost_when_it_comes_to_which/)
+
+We believe readability is enhanced by using the simplest api possible, whether we're talking about internal module function calls or standard-library macros.
+
+### `case`, `if`, and `unless`
+
+We advocate for `case` and `if` as the first tools to be considered for any control flow as they are the two simplest blocks. If a branch _can_ be expressed with an `if` statement, it _should_ be. Otherwise, `case` is the next best choice. In situations where developers might reach for an `if/elseif/else` block in other languages, `cond do` should be used.
+
+(`cond do` seems to see a paucity of use in the language, but many complex nested expressions or with statements can be improved by replacing them with a `cond do`).
+
+`unless` is a special case of `if` meant to make code read as natural-language (citation needed). While it sometimes succeeds in this goal, its absence in most programming languages often makes it feel cumbersome to programmers with non-Ruby backgrounds. Thankfully, with Styler's help developers don't need to ever reach for `unless` - expressions that are "simpler" with its use are automatically rewritten to use it.
```

Comment:
Add a comma here, I think: "Thankfully, with Styler's help, developers . . ."

Reply:
augh tough one. i could go either way hah.

Reply:
i tried it w/ the comma and it didn't work for me. gonna leave as-is